repose-aggregator/components/filters/body-patcher-filter/src/main/scala/org/openrepose/filters/bodypatcher/BodyPatcherFilter.scala

```diff
@@ -65,19 +65,12 @@ class BodyPatcherFilter @Inject()(configurationService: ConfigurationService)
         val jsonPatches: List[String] = filterJsonPatches(requestPatches)
         if (jsonPatches.nonEmpty) {
           logger.debug("Applying json patches on the request")
-          val originalValue: JsValue = Try({
-            val inputStream: ServletInputStream = httpRequest.getInputStream
-            if (inputStream.available() > 0) {
-              PJson.parse(inputStream)
-            } else {
-              logger.trace("No request body found, creating empty json")
-              PJson.obj()
-            }
-          }).recover({
-            case jpe: JsonParseException =>
-              logger.trace("Bad Json body")
-              throw new RequestBodyUnparseableException("Couldn't parse the body as json", jpe)
-          }).get
+          val originalValue: JsValue = Try(PJson.parse(httpRequest.getInputStream))
+            .recover({
+              case jpe: JsonParseException =>
+                logger.trace("Bad Json body")
+                throw new RequestBodyUnparseableException("Couldn't parse the body as json", jpe)
```

Comment:
Also i confirmed that this exception on the request side is caught and made into a 400, on the response it's intentionally allowed out as a 500, because it means the origin service isn't returning bodies that can be patched as expected.